CustomObject sync finalization (#573)
commercetools/commercetools-sync-java: docs/usage/CUSTOM_OBJECT_SYNC.md

# Custom Object Sync

Allows importing/syncing CustomObject into a commercetools project. 

> It also provides utilities for generating update actions based on the comparison of a [CustomObject](https://docs.commercetools.com/http-api-projects-custom-objects#customobject) 

Author: https://github.com/commercetools/commercetools-sync-java/pull/573/commits/144d3ebb0ef59f25ecfa945799c5f4d01f129761

against a [CustomObjectDraft](https://docs.commercetools.com/http-api-projects-custom-objects#customobjectdraft).

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Usage](#usage)
  - [Sync list of CustomObjectDrafts](#sync-list-of-customobjectdrafts)
    - [Prerequisites](#prerequisites)
    - [Running the sync](#running-the-sync)
    - [More examples of how to use the sync](#more-examples-of-how-to-use-the-sync)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Usage
        
### Sync list of CustomObjectDrafts

#### Prerequisites
1. The sync expects a list of `CustomObjectDraft`s that have their `key` and `container` fields set to be matched with

> custom objects in the target CTP project. Also, the custom objects in the target project are expected to have the `key`

Author: https://github.com/commercetools/commercetools-sync-java/pull/573/commits/144d3ebb0ef59f25ecfa945799c5f4d01f129761

# Custom Object Sync

> Allows importing/syncing CustomObject into a commercetools project. 

Reviewer: ```suggestion
Module used for importing/syncing CustomObject into a commercetools project. 
```

Author: https://github.com/commercetools/commercetools-sync-java/pull/573/commits/fdaeb1085f8ad3b4c938edbac0a5da9830bc90f0

> It also provides utilities for checking whether importing custom object is identical to existing one based on the 

Reviewer: ```suggestion
It also provides utilities for correlating a custom object to a given custom object draft based on the 
```

Author: https://github.com/commercetools/commercetools-sync-java/pull/573/commits/fdaeb1085f8ad3b4c938edbac0a5da9830bc90f0

comparison of a [CustomObject](https://docs.commercetools.com/http-api-projects-custom-objects#customobject) 
against a [CustomObjectDraft](https://docs.commercetools.com/http-api-projects-custom-objects#customobjectdraft).

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Usage](#usage)
  - [Sync list of CustomObjectDrafts](#sync-list-of-customobjectdrafts)
    - [Prerequisites](#prerequisites)
    - [Running the sync](#running-the-sync)
    - [More examples of how to use the sync](#more-examples-of-how-to-use-the-sync)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Usage
        
### Sync list of CustomObjectDrafts

#### Prerequisites
1. The sync expects a list of `CustomObjectDraft`s that have their `key` and `container` fields set to be matched with

> custom objects in the target CTP project. Therefore, the custom objects in the target project are expected to have the 

Reviewer: ```suggestion
custom objects in the target CTP project. Therefore, the custom objects in the target project are expected to have the same
```

Author: https://github.com/commercetools/commercetools-sync-java/pull/573/commits/fdaeb1085f8ad3b4c938edbac0a5da9830bc90f0

`key` and `container` fields set, otherwise they won't be matched.

2. Create a `sphereClient` [as described here](IMPORTANT_USAGE_TIPS.md#sphereclient-creation).

3. After the `sphereClient` is set up, a `CustomObjectSyncOptions` should be be built as follows:
````java
// instantiating a CustomObjectSyncOptions
final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder.of(sphereClient).build();
````

[More information about Sync Options](SYNC_OPTIONS.md).

#### Running the sync
After all the aforementioned points in the previous section have been fulfilled, to run the sync:
````java
// instantiating a CustomObjectSync
final CustomObjectSync customObjectSync = new CustomObjectSync(customObjectSyncOptions);

> // execute the sync on your list of tax categories

Reviewer: ```suggestion
// execute the sync on your list of custom object drafts
```

Author: https://github.com/commercetools/commercetools-sync-java/pull/573/commits/fdaeb1085f8ad3b4c938edbac0a5da9830bc90f0
